src/client/auth/oidc.rs

```diff
@@ -112,15 +154,93 @@ impl Callback {
             + 'static,
     {
         State {
-            callback: Self::new(callback, kind),
-            cache: Arc::new(Mutex::new(Cache::new())),
+            inner: Arc::new(Mutex::new(Some(StateInner {
+                callback: Self::new(callback, kind),
+                cache: Cache::new(),
+            }))),
+            is_user_provided: true,
+        }
+    }
+
+    /// Create azure callback.
+    #[cfg(feature = "azure-oidc")]
+    fn azure_callback(client_id: Option<&str>, resource: &str) -> StateInner {
+        use futures_util::FutureExt;
+        let resource = resource.to_string();
+        let client_id = client_id.map(|s| s.to_string());
+        let mut url = format!(
+            "http://169.254.169.254/metadata/identity/oauth2/token?api-version=2018-02-01&resource={}",
```

Comment:
I'm kind of surprised that this requires using a hardcoded insecure URI.

Reply:
right? :/